src/python/pcmdi/scripts/pcmdi_metrics_driver.py

```diff
@@ -159,13 +155,12 @@ def applyCustomKeys(O,custom_dict,var):
 
 
     #Ok at that stage we need to loop thru obs
-    dup('parameter file ref is: ',parameters.ref)
-    refs=parameters.ref
-    if isinstance(refs,list) and "all" in [x.lower() for x in refs]:
-      refs = "all"
-    if isinstance(refs,(unicode,str)):
+    dup('ref is: ',parameters.ref)
+    if isinstance(parameters.ref,list):
+        refs=parameters.ref
+    elif isinstance(parameters.ref,(unicode,str)):
         #Is it "all"
-        if refs.lower()=="all":
+        if parameters.ref.lower()=="all":
```

Comment:
this bits worries me a lot, why not using refs any longer, are you positive it's not used anywhere else?